### PR TITLE
LF: Add method to compute transaction informees.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Part of #8020.
Follow up #9416.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala [hunk 1]
```diff
@@ -279,6 +279,16 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
 
   def roots: ImmArray[Nid]
 
+  // TODO https://github.com/digital-asset/daml/issues/8020
+  //  check this make seen w.r.t. the ledger model
+  /** The informees of a all transaction.
+    */
+  def informees: Set[Ref.Party] =
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala:286]: Wondering if I should make this a `lazy val`

Comment (cocreature) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala:283]: I’d probably go for the lazy val but I don’t feel particularly strongly about this.

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala [hunk 2]
```diff
@@ -279,6 +279,13 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
 
   def roots: ImmArray[Nid]
 
+  /** The union of the informees of a all the action nodes. */
+  def informees: Set[Ref.Party] =
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala:283]:
```suggestion
  lazy val informees: Set[Ref.Party] =
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitter.scala [hunk 3]
```diff
@@ -406,24 +405,8 @@ private[kvutils] class TransactionCommitter(
...
-            accum
-        }
-
-      val allExist =
-        foldInformeeParties(transactionEntry.transaction, init = true) { (accum, party) =>
-          commitContext.get(partyStateKey(party)).fold(false)(_ => accum)
-        }
-
-      if (allExist)
+      val parties = transactionEntry.transaction.informees
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitter.scala:408]: :+1:

**Pull request merged**